nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/StandardNiFiServiceFacade.java

```diff
@@ -2939,6 +2939,115 @@ public ControllerServiceEntity updateControllerService(final Revision revision,
         return entityFactory.createControllerServiceEntity(snapshot.getComponent(), dtoFactory.createRevisionDTO(snapshot.getLastModification()), permissions, operatePermissions, bulletinEntities);
     }
 
+    @Override
+    public ControllerServiceEntity moveControllerService(final Revision revision, final ControllerServiceDTO controllerServiceDTO, final String newProcessGroupID) {
+        // get the component, ensure we have access to it, and perform the move request
+        final ControllerServiceNode controllerService = controllerServiceDAO.getControllerService(controllerServiceDTO.getId());
+        final RevisionUpdate<ControllerServiceDTO> snapshot = updateComponent(revision,
+                controllerService,
+                () -> moveControllerServiceWork(controllerService, newProcessGroupID),
+                cs -> {
+                    awaitValidationCompletion(cs);
```

Comment:
We cannot wait for validation to complete here. This is called from a web thread, and must return ASAP. There should be no need to wait for validation here, though.

Reply:
Removed call to wait for validation